elastic/kibana
# [Infrastructure UI] Filter out null bucket items from average calculation

## Summary

Closes [#152328](https://github.com/elastic/kibana/issues/152328)

This PR fixes the average calculation in the Snapshot API, filtering out buckets with null values from it, which are more likely to appear with queries that use small data ranges. 

The results after this change are equal to what Elasticsearch would calculate in the avg aggregation


### How to test 

- Make sure you have metrics data (either through enabling the system module in metricbeat or connecting your local kibana to an oblt-cli cluster)
- Navigate to `Infrastructure` > `Hosts`
- Filter the results to see a single host
- Change the data range filter and compare the KPIs against the table.
- Validate other pages that use the Snapshot API (Inventory UI and Metrics UI to see if the results there are still correct
